sktime/forecasting/tbats.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["Martin Walter"]
+__all__ = ["TBATS"]
+
+from sktime.forecasting.base._adapters import _TbatsAdapter
+from sktime.utils.check_imports import _check_soft_dependencies
+
+_check_soft_dependencies("tbats")
+
+
+class TBATS(_TbatsAdapter):
+    """TBATS estimator used to fit and select best performing model.
+    TBATS (Exponential smoothing state space model with Box-Cox
+    transformation, ARMA errors, Trigonometric Trend and Seasonal components.)
+    Model has been described in De Livera, Hyndman & Snyder (2011).
+    All of the useful methods have been implemented in parent Estimator class.
+
+    Parameters
+    ----------
+    use_box_cox: bool or None, optional (default=None)
+        If Box-Cox transformation of original series should be applied.
+        When None both cases shall be considered and better is selected by AIC.
+    box_cox_bounds: tuple, shape=(2,), optional (default=(0, 1))
+        Minimal and maximal Box-Cox parameter values.
+    use_trend: bool or None, optional (default=None)
+        Indicates whether to include a trend or not.
+        When None both cases shall be considered and better is selected by AIC.
+    use_damped_trend: bool or None, optional (default=None)
+        Indicates whether to include a damping parameter in the trend or not.
+        Applies only when trend is used.
+        When None both cases shall be considered and better is selected by AIC.
+    seasonal_periods: iterable or array-like of floats, optional (default=None)
+        Length of each of the periods (amount of observations in each period).
+        Accepts int and float values here.
+        When None or empty array, non-seasonal model shall be fitted.
+    use_arma_errors: bool, optional (default=True)
+        When True BATS will try to improve the model by modelling residuals with ARMA.
+        Best model will be selected by AIC.
+        If False, ARMA residuals modeling will not be considered.
+    show_warnings: bool, optional (default=True)
+        If warnings should be shown or not.
+        Also see Model.warnings variable that contains all model related warnings.
+    n_jobs: int, optional (default=None)
+        How many jobs to run in parallel when fitting BATS model.
+        When not provided BATS shall try to utilize all available cpu cores.
+    multiprocessing_start_method: str, optional (default='spawn')
+        How threads should be started. See also:
+        https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
+    context: abstract.ContextInterface, optional (default=None)
+        For advanced users only. Provide this to override default behaviors
+    """
+
+    def __init__(self, **kwargs):
```

Comment:
Having `**kwargs` in the constructor is usually not a good idea - makes it hard for users to understand.

Reply:
This `**kwargs` are required here because some parts of the `__init__` are defined in the parent `_TbatsAdapter` and one of the `self` arguments is defined in the children. I dont know any other way to do that and this is the most minimal way. Otherwise I would have to keep the whole `__init__` in the children and do copy paste.

Reply:
I'd say that's preferable to having `*kwargs`. 

Could the `_ModelClass` not be a class variable so that we don't have to overwrite the `init` at all if it has the same signature for both methods? Something like this:
```python
class TBATS(_TbatsAdapter):
    _ModelClass = _TBATS
```

Reply:
it has to be `self._ModelClass` because in `fit()` I am doing then `self._instantiate_model()` which requires to read then `self._ModelClass`

Reply:
Yes but class variables are accessible to objects via `self`, no? This for example works:

```python
class A:
    _x = 1

    def get_x(self):
        return self._x

a = A()
a.get_x()
>>> 1
```

Reply:
I just tried this, but doing to I would have to do `_ModelClass = TBATS` in `tbats.py` but this raises an issue `local variable '_ModelClass ' is assigned to but never used flake8(F841)`

Reply:
Ok I got it resolved, now its even more minimalistic, probably the most minimal wrapper out there in the ML world :D So lets got for it!